src/expressions/include/antares/expressions/visitors/TimeIndexVisitor.h

```diff
@@ -38,13 +40,18 @@ class TimeIndexVisitor: public NodeVisitor<TimeIndex>
      *
      * @param context The context containing the time index for each node.
      */
-    explicit TimeIndexVisitor(std::unordered_map<const Nodes::Node*, TimeIndex> context);
-    explicit TimeIndexVisitor() = default;
+    explicit TimeIndexVisitor(
+      std::unordered_map<const Nodes::Node*, TimeIndex> context,
```

Comment:
is this ctor used?

Reply:
no, it's unused. 
There's a TODO about removing it. I will remove it.

Reply:
No, you're right, it's not used in the production code.
It's used only in unit tests (see **test_TimeIndexVisitor.cpp**)
This constructor initialize a data member `context_`.
I'm not sure of the utility of this data members, apart from testing purpose.

My opinion is we should remove this data member, replace it's use with a exception throw, and finally remove the constructor and the unit tests using it.